parlai/core/dialog_teacher.py

```diff
@@ -322,7 +322,9 @@ def __len__(self):
         length_file = self.datafile + ".length"
         if not os.path.isfile(length_file):
             length = 0
-            for episode in self._read_episode(self.data_loader(self.datafile)):
+            for episode in self.data:
+                if episode == -1:
+                    break
```

Comment:
wait--will this prevent it from iterating again? try this with validation data or something. may need to reset after this.